cub/test/catch2_test_block_load.cu

```diff
@@ -77,50 +68,45 @@ __global__ void kernel(std::integral_constant<bool, true>, InputIteratorT input,
 
   for (int i = 0; i < ItemsPerThread; i++)
   {
-    const int idx = output_idx<ItemsPerThread, ThreadsInBlock, LoadAlgorithm>::get(i);
-
+    const int idx = get_output_idx<ItemsPerThread, ThreadsInBlock, LoadAlgorithm>(i);
     if (idx < num_items)
     {
       output[idx] = data[i];
     }
   }
 }
 
-template <typename InputIteratorT,
-          typename OutputIteratorT,
-          int ItemsPerThread,
+template <int ItemsPerThread,
           int ThreadsInBlock,
-          cub::BlockLoadAlgorithm /* LoadAlgorithm */>
-__global__ void kernel(std::integral_constant<bool, false>, InputIteratorT input, OutputIteratorT output, int num_items)
+          cub::BlockLoadAlgorithm /* LoadAlgorithm */,
+          typename InputIteratorT,
+          typename OutputIteratorT>
+__global__ void kernel(cuda::std::false_type, InputIteratorT input, OutputIteratorT output, int num_items)
 {
   for (int i = 0; i < ItemsPerThread; i++)
   {
-    const int idx = output_idx<ItemsPerThread, ThreadsInBlock, cub::BlockLoadAlgorithm::BLOCK_LOAD_DIRECT>::get(i);
-
+    const int idx = get_output_idx<ItemsPerThread, ThreadsInBlock, cub::BlockLoadAlgorithm::BLOCK_LOAD_DIRECT>(i);
     if (idx < num_items)
     {
       output[idx] = input[idx];
     }
   }
 }
 
-template <int ItemsPerThread,
-          int ThreadsInBlock,
-          cub::BlockLoadAlgorithm LoadAlgorithm,
-          typename InputIteratorT,
-          typename OutputIteratorT>
-void block_load(InputIteratorT input, OutputIteratorT output, int num_items)
+template <int ItemsPerThread, int ThreadsInBlock, cub::BlockLoadAlgorithm LoadAlgorithm, typename T, typename InputIteratorT>
+void test_block_load(const c2h::device_vector<T>& d_input, InputIteratorT input)
 {
-  using input_t                       = cub::detail::value_t<InputIteratorT>;
-  using block_load_t                  = cub::BlockLoad<input_t, ThreadsInBlock, ItemsPerThread, LoadAlgorithm>;
-  using storage_t                     = typename block_load_t::TempStorage;
-  constexpr bool sufficient_resources = sizeof(storage_t) <= cub::detail::max_smem_per_block;
-
-  kernel<InputIteratorT, OutputIteratorT, ItemsPerThread, ThreadsInBlock, LoadAlgorithm>
-    <<<1, ThreadsInBlock>>>(std::integral_constant<bool, sufficient_resources>{}, input, output, num_items);
+  using block_load_t = cub::BlockLoad<T, ThreadsInBlock, ItemsPerThread, LoadAlgorithm>;
+  using storage_t    = typename block_load_t::TempStorage;
+  constexpr auto sufficient_resources =
+    cuda::std::bool_constant<sizeof(storage_t) <= cub::detail::max_smem_per_block>{};
 
+  c2h::device_vector<T> d_output(d_input.size());
+  kernel<ItemsPerThread, ThreadsInBlock, LoadAlgorithm>
+    <<<1, ThreadsInBlock>>>(sufficient_resources, input, thrust::raw_pointer_cast(d_output.data()), d_input.size());
```

Comment:
Leaving this here so you don't have to go through the build logs: MSVC build fails due to implicit conversion from `size_t` to `int`.
```suggestion
    <<<1, ThreadsInBlock>>>(sufficient_resources, input, thrust::raw_pointer_cast(d_output.data()), static_cast<int>(d_input.size()));
```